File: json/ocaml/update/json_update.ml
(* ------------------------------------------------------------------- *)
(*
	json_update.ml

					Apr/09/2013
*)
(* ------------------------------------------------------------------- *)
open	Printf
open	Json_type
open	Json_type.Browse

open	Text_manipulate
open	Json_manipulate
(* ------------------------------------------------------------------- *)
let json_file = Sys.argv.(1)
let key = Sys.argv.(2)
let population_in = int_of_string Sys.argv.(3)
let _ =	printf "%s\n" key
let _ =	printf "%d\n" population_in

let json_str = file_to_str_proc json_file
let dict_aa = Hashtbl.create 100
let _ = json_to_dict_proc json_str dict_aa
let dict_bb = dict_update_proc dict_aa key population_in



(*
let _ = dict_update_proc dict_aa key population_in
let _ = Hashtbl.iter (fun key value -> print_record key dict_aa) dict_aa
*)


(*
let json_obj_bb = Json_type.Build.object dict_aa
let _ = Json_io.save_json "tmp001.json" dict_aa
*)

let _ = print_endline "*** 終了 ***"
(* ------------------------------------------------------------------- *)

File: json/typescript/update/json_update.ts
// ---------------------------------------------------------------
//	json_update.ts
//
//					Jul/28/2020
//
// ---------------------------------------------------------------
var fs = require("fs")
var text_manipulate = require('/var/www/data_base/common/typescript_common/text_manipulate')

//
console.error ("*** 開始 ***")
var filename:string = process.argv[2]
var id_in:string = process.argv[3]
var population_in:number = parseInt(process.argv[4])

console.log (filename)
console.log (id_in + "\t" + population_in)

const json_str:string = fs.readFileSync(filename,'utf8')
var dict_aa:Object = JSON.parse (json_str)

const dict_bb = text_manipulate.dict_update_proc (dict_aa,id_in,population_in)

const json_str_out:string = JSON.stringify(dict_bb)

fs.writeFile (filename,json_str_out, function (err)
	{
	if (err) {
		console.log("Error on write: " + err)
		} else {
		console.log("File written.")
		}
	})

console.error ("*** 終了 ***")

// ---------------------------------------------------------------
